### PR TITLE
Add various fns

### LINKED ISSUE #102: Support for Fused Multiply-Add (FMA)
From my limited testing, multiplying and then adding `SimdF32`/`SimdF64` does not result in FMA instructions, and rightly so, however it would be useful to have a method on these types and a corresponding platform intrinsic to generate LLVM calls to `llvm.fma.*` intrinsics, which support vector types.

Edit: In fact it looks like the `simd_fma` platform intrinsic is already available, just not used by the crate?

### PR DESCRIPTION
**Pull request by calebzulawski**
Adds
* `Sum`/`Product` traits
* `recip`/`to_degrees`/`to_radians`/`min`/`max`/`clamp`/`signum`/`copysign` (#14)
* `mul_add` (#14, fixes #102)

### CODE REVIEW

File: crates/core_simd/src/vector/float.rs [hunk 1]
```diff
@@ -36,13 +36,44 @@ macro_rules! impl_float_vector {
...
+            #[inline]
+            pub fn to_degrees(self) -> Self {
+                // to_degrees uses a special constant for better precision, so extract that constant
+                self * Self::splat($type::to_degrees(1.))
+            }
+
+            /// Converts each lane from degrees to radians.
+            #[inline]
+            pub fn to_radians(self) -> Self {
+                self * Self::splat(core::$type::consts::PI / 180.)
```
Comment (programmerjake) [inline crates/core_simd/src/vector/float.rs:75]: this could also use a special constant for more precision

Comment (miguelraz) [reply crates/core_simd/src/vector/float.rs:75]: I'd hope the division gets optimized to a fmul here, otherwise that would be bad.

Comment (calebzulawski) [reply crates/core_simd/src/vector/float.rs:75]: It should be optimized to a constant.

**Comment by workingjubilee** [comment]
Hello! The question arose in the conversation of: "Is this ready to merge?"

**Comment by calebzulawski** [comment]
Yep!

**Pull request merged**